Communication/test/com/gymfox/communication/IPv4AddressTest.java

```diff
@@ -0,0 +1,100 @@
+package com.gymfox.communication;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class IPv4AddressTest {
+    @Test
+    public void equalsIpAsStringTest() {
+        Assert.assertEquals("255.255.255.255", new IPv4Address("255.255.255.255").getIpString());
+        Assert.assertEquals("127.12.45.22", new IPv4Address("127.12.45.22").getIpString());
+        Assert.assertEquals("189.11.23.211", new IPv4Address("189.11.23.211").getIpString());
+    }
+
+    @Test
+    public void equalsIpAsLongWithStringTest() {
+        Assert.assertEquals(2131504406l, new IPv4Address("127.12.45.22").getIpLong());
+        Assert.assertEquals(3171620819l, new IPv4Address("189.11.23.211").getIpLong());
+        Assert.assertEquals(4294967295l, new IPv4Address("255.255.255.255").getIpLong());
+    }
+
+    @Test
+    public void equalsIpAsLongTest() {
+        Assert.assertEquals(2131504406l, new IPv4Address(2131504406l).getIpLong());
```

Comment:
Minor: regarding that you use single value twice, it would be better to initialize variable to keep the value there:

`long expectedLongValue1 = 2131504406l;`
`Assert.assertEquals(expectedLongValue1, new IPv4Address(expectedLongValue1).getIpLong());`